x/tokenfactory/types/tx.go

```diff
@@ -32,9 +32,11 @@ func (msg *MsgUpdateParams) Validate() error {
 		return errorsmod.Wrap(err, "authority is invalid")
 	}
 
+	// TODO: This is inconsistent. Per Params.Validate() an empty creator address is valid as long as
+	// DenomCreationFee is nil. But This check fails if FeeCollectorAddress is unset.
 	if _, err := sdk.AccAddressFromBech32(msg.Params.FeeCollectorAddress); err != nil {
 		return errorsmod.Wrap(err, "fee_collector_address is invalid")
 	}
```

Comment:
why do we even need this piece of code? why there is a part of `msg.Params` validation placed here, not in the `msg.Params.Validate`? we literally have the same validation inside the `msg.Params.Validate` method:
<img width="605" alt="image" src="https://github.com/neutron-org/neutron/assets/34917380/1da66de6-1e15-4ad2-bf75-c46035de7823">

Reply:
ValidateFeeCollectorAddress allows an empty address. But params.Validate() does not. I'm fairly certain we should remove the validation of FeeCollectorAddress in params.Validate(). But doing so breaks some tests. So in the interest of keeping the commit as clear and atomic as possible, I did not want to change anything that wasn't essential.